server/gc/src/main/java/org/apache/accumulo/gc/GarbageCollectionAlgorithm.java

```diff
@@ -217,11 +221,50 @@ private void confirmDeletes(GarbageCollectionEnvironment gce,
         throw new RuntimeException(
             "Scanner over metadata table returned unexpected column : " + entry.getKey());
     }
+    Set<String> tableIdsAfter = gce.getTableIDs();
+    ensureAllTablesChecked(Collections.unmodifiableSet(tableIdsBefore),
+        Collections.unmodifiableSet(tableIdsSeen), Collections.unmodifiableSet(tableIdsAfter));
 
     confirmDeletesFromReplication(gce.getReplicationNeededIterator(),
         candidateMap.entrySet().iterator());
   }
 
+  @VisibleForTesting
+  /**
+   *
```

Comment:
some comments here might be nice describing the context of this check.  The context includes that we pulled the candidate delete list prior to these scans.  Hence we only really care that the tableIdsSeen is a super set of the intersection of the ids before and after.  There need to be more words to connect those two sentences....